File: minibench/benchmark.rb

start_time = Time.now

foo = 0
bar = ''

1000000.times do |i|
	foo += i / 10.0
	if((foo.floor() % 3) == 2)
		foo -= (i - (2.0 + Math.sqrt(i)).round)
		bar.concat(i.to_s)
	end
end

end_time = Time.now
diff = end_time - start_time

puts sprintf("%.6f seconds\n", diff)

File: minibench/benchmark.py

import time
import math

start = time.time()

foo = 0
bar = ''
for i in range(0, 1000000):
	foo += i / 10
	if (math.floor(foo) % 3) == 2:
		foo -= (i - round(2.0 + math.sqrt(i)))
		bar += str(i)

end = time.time()
diff = end - start

print("%.6f seconds" % diff)
